frontend/src/Metamaps/Views/TopicCard.js

```diff
@@ -5,69 +5,60 @@ import ReactDOM from 'react-dom'
 
 import Active from '../Active'
 import Visualize from '../Visualize'
-import GlobalUI from '../GlobalUI'
-
-import ReactTopicCard from '../../components/TopicCard'
+import GlobalUI, { ReactApp } from '../GlobalUI'
 
 const TopicCard = {
-  openTopicCard: null, // stores the topic that's currently open
+  openTopic: null, // stores the topic that's currently open
   metacodeSets: [],
+  updateTopic: () => {},
+  onTopicFollow: () => {},
+  redrawCanvas: () => {
+    Visualize.mGraph.plot()
+  },
   init: function(serverData) {
     const self = TopicCard
     self.metacodeSets = serverData.metacodeSets
   },
   populateShowCard: function(topic) {
     const self = TopicCard
-    ReactDOM.render(
-      React.createElement(ReactTopicCard, {
-        topic: topic,
-        ActiveMapper: Active.Mapper,
-        updateTopic: obj => {
-          topic.save(obj, { success: topic => self.populateShowCard(topic) })
-        },
-        onFollow: () => {
-          const isFollowing = topic.isFollowedBy(Active.Mapper)
-          $.post({
-            url: `/topics/${topic.id}/${isFollowing ? 'un' : ''}follow`
-          })
-          if (isFollowing) {
-            GlobalUI.notifyUser('You are no longer following this topic')
-            Active.Mapper.unfollowTopic(topic.id)
-          } else {
-            GlobalUI.notifyUser('You are now following this topic')
-            Active.Mapper.followTopic(topic.id)
-          }
-          self.populateShowCard(topic)
-        },
-        metacodeSets: self.metacodeSets,
-        redrawCanvas: () => {
-          Visualize.mGraph.plot()
-        }
-      }),
-      document.getElementById('showcard')
-    )
-
+    TopicCard.updateTopic = obj => {
+      topic.save(obj, { success: topic => self.populateShowCard(topic) })
+    }
+    TopicCard.onTopicFollow = () => {
+      const isFollowing = topic.isFollowedBy(Active.Mapper)
+      $.post({
+        url: `/topics/${topic.id}/${isFollowing ? 'un' : ''}follow`
+      })
+      if (isFollowing) {
+        GlobalUI.notifyUser('You are no longer following this topic')
+        Active.Mapper.unfollowTopic(topic.id)
+      } else {
+        GlobalUI.notifyUser('You are now following this topic')
+        Active.Mapper.followTopic(topic.id)
+      }
+      self.populateShowCard(topic)
+    }
```

Comment:
hmm, this is pretty clever but maybe too clever. is there a simpler way to do it that makes it clearer what's going on?